src/pages/index.js

```diff
@@ -22,128 +23,270 @@ import {
-  // profileDescriptionInput.value = userInfo.description;
+  console.log(`Fill Profile Form: userInfo`);
+  console.log(userInfo);
+
+  // reformat data (change about property to description)
+  const updatedData = { name: userInfo.name, description: userInfo.about };
+  profilePopup.setInputValues(updatedData);
+}
+
+function openEditAvatarModal() {
+  editAvatarPopup.open();
 }
 
-// Button Handlers
 function openEditProfileModal() {
   fillProfileForm();
   formValidators[profileForm.getAttribute("id")].resetValidation();
 
   profilePopup.open();
 }
 
+function openImageModal(data) {
+  imgagePopup.open(data);
+}
+
 function openAddCardModal() {
   addCardPopup.open();
 }
 
-// Button Event Listeners
+function createCard(data) {
+  const cardElement = new Card(
+    data,
+    cardTemplate,
+    openImageModal,
+    openDeleteModal,
+    handleLikeButton
+  );
+
+  return cardElement.getView();
+}
+
+function openDeleteModal(currentCardId, currentCardElement) {
+  deleteCardPopup.open();
+  deleteCardPopup.currentCardId = currentCardId;
+  deleteCardPopup.currentCardElement = currentCardElement;
+}
+
+function handleCardDelete() {
+  return api
+    .deleteCard(deleteCardPopup.currentCardId)
+    .then(deleteCardPopup.currentCardElement.remove())
+    .catch(console.error);
+}
+
+function handleLikeButton(cardId, isLiked) {
+  console.log(cardId);
+  console.log(isLiked);
+
+  if (!isLiked) {
+    return api
+      .addLike(cardId)
+      .then(() => true)
+      .catch(console.error);
+  } else {
+    return api
+      .removeLike(cardId)
+      .then(() => false)
+      .catch(console.error);
+  }
+}
+
+//// Button Event Listeners ////
+// avatar image click
+profileAvatarWrapper.addEventListener("click", openEditAvatarModal);
+
 // profile edit button click
 profileEditButton.addEventListener("click", openEditProfileModal);
 
 // add card button click
 addCardButton.addEventListener("click", openAddCardModal);
 
-//// Forms ////
-const profileForm = document.forms["edit-profile-form"];
-const addCardForm = document.forms["add-card-form"];
+//// Form Submit Handlers ////
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
 
-//// Form Input Fields ////
-const profileNameInput = profileForm.querySelector(".modal__input_type_name");
-const profileDescriptionInput = profileForm.querySelector(
-  ".modal__input_type_description"
-);
-const newCardTitleInput = addCardForm.querySelector(".modal__input_type_title");
-const newCardUrlInput = addCardForm.querySelector(".modal__input_type_url");
+// Avatar form submission handler
+function handleAvatarFormSubmit(data) {
+  function makeRequest() {
+    return api.updateUserAvatar(data).then((profileAvatar.src = data.link));
```

Comment:
`(profileAvatar.src = data.link)`

Class `UserInfo` should set the entire user’s data (including the **avatar**)